p2p/src/transport.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt;
+use std::io::{Read, Write};
+use std::net::SocketAddr;
+
+use tendermint::public_key::PublicKey;
+
+use eyre::Result;
+
+pub struct BindInfo {
+    pub addr: SocketAddr,
+    pub advertise_addrs: Vec<SocketAddr>,
+    pub public_key: PublicKey,
```

Comment:
I don't think public_key is enough. `SecretConnection` needs `private_key`

Reply:
Which type would you suggest to use here?

Reply:
```rust
    pub private_key: ed25519::Keypair,
```

?